File: RB101/small_problems/easy_7/6.rb
def staggered_case(string)
  idx = 0
  string.chars.map do |char|
    if !!/[a-zA-Z]/.match(char)
      idx += 1
      idx.odd? ? char.upcase : char.downcase
    else
      char
    end
  end.join
end

p staggered_case('I Love Launch School!') == 'I lOvE lAuNcH sChOoL!'
p staggered_case('ALL CAPS') == 'AlL cApS'
p staggered_case('ignore 77 the 444 numbers') == 'IgNoRe 77 ThE 444 nUmBeRs'

File: JS210/small_problems/strings/6.js
//let staggeredCase = str => str.split('').map((chr, i) => i % 2 == 0 ? chr.toUpperCase() : chr.toLowerCase()).join('');

function staggeredCase(str) {
  return str.split('')
            .map((chr, i) => i % 2 == 0 ? chr.toUpperCase() : chr.toLowerCase())
            .join('');
}

console.log(staggeredCase('I Love Launch School!'));        // "I LoVe lAuNcH ScHoOl!"
console.log(staggeredCase('ALL_CAPS'));                     // "AlL_CaPs"
console.log(staggeredCase('ignore 77 the 4444 numbers'));   // "IgNoRe 77 ThE 4444 nUmBeRs"